base/shell/explorer/shellservice.cpp

```diff
@@ -126,6 +126,7 @@ HRESULT InitShellServices(HDPA * phdpa)
     return count > 0 ? S_OK : S_FALSE;
 
 cleanup:
+    RegCloseKey(hkey);
```

Comment:
Not right: check lines 118 and 123...

Reply:
@SergeGautherie, I believe that Line 118 just needs to be deleted after the RegCloseKey() is inserted under the Cleanup: label. Or looking at it another way the RegCloseKey() in Line 118 needs to be moved under the Cleanup: label. Thanks.

Reply:
@Doug-Lyons, no: check line 126.

Reply:
@SergeGautherie, You are correct. Also, Line #126 should probably be as follows:
hr = count > 0 ? S_OK : S_FALSE;
Do you agree that these three changes should handle all of the obvious conditions now?

Reply:
@Doug-Lyons, no...

Reply:
OK 118 just need to come after 123. Nice catch @SergeGautherie